Content.Server/Body/Systems/LungSystem.cs

```diff
@@ -50,175 +42,27 @@ private void OnGotEquipped(EntityUid uid, BreathToolComponent component, GotEqui
+        for (int i = 0; i < Atmospherics.TotalNumberOfGases; i++)
         {
-            case LungStatus.Inhaling:
-                Inhale(uid, absoluteTime);
-                lung.Status = LungStatus.Exhaling;
-                break;
-            case LungStatus.Exhaling:
-                Exhale(uid, absoluteTime);
-                lung.Status = LungStatus.Inhaling;
-                break;
-            default:
-                throw new ArgumentOutOfRangeException();
+            var moles = lung.Air.Moles[i];
+            if (moles <= 0)
+                continue;
+            var reagent = _atmosphereSystem.GasReagents[i];
+            if (reagent == null) continue;
+
+            var amount = moles * Atmospherics.BreathMolesToReagentMultiplier;
+            _solutionContainerSystem.TryAddReagent(uid, lung.LungSolution, reagent, amount, out _);
         }
 
-        lung.AccumulatedFrametime = absoluteTime - delay;
-    }
-
-    /// <summary>
-    ///     Tries to find an air mixture to inhale from, then inhales from it.
-    /// </summary>
-    public void Inhale(EntityUid uid, float frameTime,
-        LungComponent? lung=null,
-        SharedMechanismComponent? mech=null)
-    {
-        if (!Resolve(uid, ref lung, ref mech))
-            return;
-
-        // TODO Jesus Christ make this event based.
-        if (mech.Body != null &&
-            EntityManager.TryGetComponent((mech.Body).Owner, out InternalsComponent? internals) &&
-            internals.BreathToolEntity != null &&
-            internals.GasTankEntity != null &&
-            EntityManager.TryGetComponent(internals.BreathToolEntity, out BreathToolComponent? breathTool) &&
-            breathTool.IsFunctional &&
-            EntityManager.TryGetComponent(internals.GasTankEntity, out GasTankComponent? gasTank))
-        {
-            TakeGasFrom(uid, frameTime, gasTank.RemoveAirVolume(Atmospherics.BreathVolume), lung);
-            return;
-        }
-
-        if (_atmosSys.GetTileMixture(EntityManager.GetComponent<TransformComponent>(uid).Coordinates, true) is not { } tileAir)
-        {
-            return;
-        }
-
-        TakeGasFrom(uid, frameTime, tileAir, lung);
-    }
-
-    /// <summary>
-    ///     Inhales directly from a given mixture.
-    /// </summary>
-    public void TakeGasFrom(EntityUid uid, float frameTime, GasMixture from,
-        LungComponent? lung=null,
-        SharedMechanismComponent? mech=null)
-    {
-        if (!Resolve(uid, ref lung, ref mech))
-            return;
-
-        var ratio = (Atmospherics.BreathVolume / from.Volume) * frameTime;
-
-        _atmosSys.Merge(lung.Air, from.RemoveRatio(ratio));
-
-        // Push to bloodstream
-        if (mech.Body == null)
-            return;
-
-        if (!EntityManager.TryGetComponent((mech.Body).Owner, out BloodstreamComponent? bloodstream))
-            return;
-
-        var to = bloodstream.Air;
-
-        _atmosSys.Merge(to, lung.Air);
         lung.Air.Clear();
     }
```

Comment:
If I'm not missing anything, this `Clear()` means that only gasses that turn into reagents which metabolise into gasses will be exhaled again. So currently the lung will only ever exhale CO2, and any inert gasses like nitrogen are just destroyed, right?

I guess the `clear()` should be removed and any gasses that turn into reagent just need to be directly removed from the air mixture?

Reply:
yeah that makes more sense